qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/database/DatabaseSession.groovy

```diff
@@ -20,45 +16,44 @@ import java.sql.SQLException
  * @author: Jennifer Bödker
  *
  */
+@Log4j2
 class DatabaseSession {
 
     protected static DatabaseSession INSTANCE
 
     private BasicDataSource dataSource
 
-    private static final Logger LOG = LogManager.getLogger(DatabaseSession.class)
-
-    //create session with DatabaseSession.init() and then DatabaseSession.INSTANCE to use the created session
-    private DatabaseSession(String user, String password, String host, String port, String sqlDatabase) {
-
-        def url = "jdbc:mysql://" + host + ":" + port + "/" + sqlDatabase
-
-        dataSource = new BasicDataSource()
-        dataSource.setUrl(url)
-        dataSource.setUsername(user)
-        dataSource.setPassword(password)
-        dataSource.setMinIdle(5)
-        dataSource.setMaxIdle(10)
-        dataSource.setMaxOpenPreparedStatements(100)
-
-        LOG.info("MySQL Database instance created")
+    private DatabaseSession() {
+        //This is a private Singleton constructor
+        dataSource = null
```

Comment:
What is the reason for changing this? I think we did not discuss this in our meetings, did we? I think it's nice to see you are motivated to clean up code and advance it but I also think we should stick to what was actually intended by this PR. Changing the database session code was not necessary imo.

Reply:
Singleton pattern is the reason. This was only partially followed here and I changed it to match the common pattern.